distribution/src/bin/elasticsearch-env.bat

```diff
@@ -57,8 +57,5 @@ set ES_DISTRIBUTION_FLAVOR=${es.distribution.flavor}
 set ES_DISTRIBUTION_TYPE=${es.distribution.type}
 
 if not defined ES_TMPDIR (
-  set ES_TMPDIR=!TMP!\elasticsearch
-  if not exist "!ES_TMPDIR!" (
-    mkdir "!ES_TMPDIR!"
-  )
+  for /f "tokens=* usebackq" %%a in (`"%JAVA% -cp "!ES_CLASSPATH!" "org.elasticsearch.tools.launchers.Mktemp""`) do set ES_TMPDIR=%%a
```

Comment:
This will create clutter on Windows machines that we weren't previously creating.  On Windows `%TMP%` defaults to a secure directory that is only accessible by one user, so there's no security need to create a unique sub-directory.  The only reason for doing it is consistency with Linux and macOS.

On macOS and Linux we create clutter in the temp directory but macOS cleans up on reboot and Linux cleans up after 10 days so these directories do eventually get removed.  I think on Windows the temp directory is never automatically cleaned, so if somebody repeatedly starts and stops Elasticsearch then these directories will build up forever.

So it's a question of whether consistency between platforms is so important that it's worth dealing with the complaints that we're cluttering people's Windows temp directories.  If we'd rather avoid the complaints then this batch file could stay as it was.

Also, if the consensus is that consistency between platforms is the most important thing and we do change it, might it affect the Windows installer @russcam and @Mpdreamz?

Reply:
You raise a good point @droberts195, I pushed 7967b75dd6613fb633f5d49dd05e70d5294a2d56. What do you think?